src/databricks/labs/ucx/source_code/notebooks/magic.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+import re
+import shlex
+
+from astroid import Call, Name, Const, NodeNG  # type: ignore
+
+from databricks.labs.ucx.source_code.graph import DependencyGraph, DependencyProblem
+from databricks.labs.ucx.source_code.linters.imports import ProblemFactory
+from databricks.labs.ucx.source_code.linters.python_ast import NodeBase, Tree
+
+
+class MagicCommand(NodeBase):
+
+    @classmethod
+    def convert_magic_lines_to_magic_commands(cls, python_code: str):
+        lines = python_code.split("\n")
+        for i, line in enumerate(lines):
+            if not line.startswith("%"):
+                continue
+            lines[i] = f"magic_command({line.encode()!r})"
```

Comment:
This `magic_command` "method" is added to use `python_ast.tree` I understand?

Reply:
not exactly, we're using astroid, and we need magic commands to be converted to regular python code